### PR TITLE
Fix flow type errors of all viewers

### PR DESCRIPTION
**Pull request by btzr-io**
Hacktober fest contribution: Flow type errors #2024

### CODE REVIEW

**Comment by btzr-io** [comment]
@seanyesmunt I still don't know how handle `react.createRef` errors, any ideas?
https://flow.org/en/docs/react/refs/

File: src/renderer/component/viewers/threeViewer/index.jsx [hunk 1]
```diff
@@ -187,7 +193,37 @@ class ThreeViewer extends React.PureComponent<Props, State> {
     }
   }
 
-  transformGroup(group) {
+  // Define component types
+  theme: viewerTheme;
+  themes: { dark: viewerTheme, light: viewerTheme };
+  materialColor: string;
+  // Too complex to add a custom type
+  // Posible solution: https://www.npmjs.com/package/@types/three
```
Comment (btzr-io) [inline src/renderer/component/viewers/threeViewer/index.jsx]: Should we use this instead of `any`?
 https://www.npmjs.com/package/@types/three

Comment (seanyesmunt) [reply src/renderer/component/viewers/threeViewer/index.jsx]: Nice. Definitely

Comment (btzr-io) [reply src/renderer/component/viewers/threeViewer/index.jsx]: Looks like they are `TypeScript` type definitions :disappointed:

Comment (btzr-io) [reply src/renderer/component/viewers/threeViewer/index.jsx]: I can't find anything for flow :stuck_out_tongue:

**Comment by seanyesmunt** [comment]
@btzr-io I'm not sure. You can just stick with assigning it in the callback

```
// constructor
this.textArea = null

// render
<textarea ref={input => this.textArea = input} />

```

File: src/renderer/component/viewers/threeViewer/index.jsx (hunk 1 above)
Comment (btzr-io) [reply src/renderer/component/viewers/threeViewer/index.jsx]: I can't use / import it:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/three/index.d.ts

Comment (seanyesmunt) [reply src/renderer/component/viewers/threeViewer/index.jsx]: I gotcha. Oh well.

**Comment by btzr-io** [comment]
@chrisza4 I keep getting this errors:
```
Cannot call React.createRef because property createRef is missing in object type [1].

     src/renderer/component/viewers/codeViewer.jsx
      29| class CodeViewer extends React.PureComponent<Props> {
      30|   constructor(props: Props) {
      31|     super(props);
      32|     this.textarea = React.createRef();
      33|     this.codeMirror = null;
      34|   }
      35|
```
```
Cannot create textarea element because a callable signature is missing in object type [1] but exists in function
type [2] in property ref.

     src/renderer/component/viewers/codeViewer.jsx
 [1]  57|   textarea: { current: null | HTMLTextAreaElement };
      58|   codeMirror: any;
      59|
      60|   render() {
      61|     const { value } = this.props;
      62|     return (
      63|       <div className="code-viewer" onContextMenu={stopContextMenu}>
      64|         <textarea ref={this.textarea} disabled value={value} />
      65|       </div>
      66|     );
      67|   }

```

**Pull request merged**